### PR TITLE
fix: advanced marker anchoring

### PR DESCRIPTION
**Pull request by mrMetalWood**
First draft for improved and more robust advanced marker anchoring. Still needs some testing...

Main change is moving away from the "0, 0" div for the anchoring reset. That seemed to cause problems in a few situations

This should address and fix:
- #569 
- #559
- #542 
- #560

### CODE REVIEW

File: src/components/__tests__/advanced-marker.test.tsx [hunk 1]
```diff
@@ -148,8 +148,10 @@ describe('map and marker-library loaded', () => {
       .get(google.maps.marker.AdvancedMarkerElement)
       .at(0) as google.maps.marker.AdvancedMarkerElement;
 
-    expect(marker.content?.firstChild).toHaveClass('classname-test');
-    expect(marker.content?.firstChild).toHaveStyle('width: 200px');
+    expect(marker.content?.firstChild?.firstChild).toHaveClass(
```
Comment (usefulthink) [inline src/components/__tests__/advanced-marker.test.tsx:151]: wondering if we should replace this with something like "expect that there is an element with classname-test somewhere in marker.content" – feels like specifying the exact hierarchy here isn't actually what we want to test.

File: src/components/advanced-marker.tsx [hunk 2]
```diff
@@ -31,6 +31,10 @@ export function isAdvancedMarker(
   );
 }
 
+function isElementNode(node: Node): node is HTMLElement {
+  return (node as Node).nodeType === Node.ELEMENT_NODE;
```
Comment (usefulthink) [inline src/components/advanced-marker.tsx:35]: isn't `as Node` redundant here, since that's the type given above?

**Review by usefulthink — COMMENTED** [review]
Looking really good! 

What I would really like to have still is an overview of which elements are created, when and why, and how this all works together. Not sure if that has to be in the documentation or just somewhere in the code. We could also tackle this in a follow-up.

File: src/components/advanced-marker.tsx (hunk 2 above)
Comment (mrMetalWood) [reply src/components/advanced-marker.tsx:35]: Ah yes, you're right.

File: src/components/__tests__/advanced-marker.test.tsx (hunk 1 above)
Comment (mrMetalWood) [reply src/components/__tests__/advanced-marker.test.tsx:151]: Yes, that sounds like a better approach.

**Comment by mrMetalWood** [comment]
> Looking really good!
> 
> What I would really like to have still is an overview of which elements are created, when and why, and how this all works together. Not sure if that has to be in the documentation or just somewhere in the code. We could also tackle this in a follow-up.

Agreed, that would be helpful.

**Review by usefulthink — APPROVED** [review]

**Pull request merged**